core/src/com/unciv/ui/civilopedia/CivilopediaScreen.kt

```diff
@@ -159,31 +179,35 @@ class CivilopediaScreen(
                     CivilopediaEntry(
                         it.name,
                         it.getDescription(false, null, ruleset),
-                        CivilopediaCategories.Wonder.getImage?.invoke(it.name, imageSize)
+                        CivilopediaCategories.Wonder.getImage?.invoke(it.name, imageSize),
+                        (it as? ICivilopediaText).takeUnless { ct -> ct==null || ct.isCivilopediaTextEmpty() }
```

Comment:
This line repeats itself, couldn't the CivilopediaEntry constructor handle this?

Reply:
It's nearing the point where we would expect a `CivilopediaCategories.values().forEach{}`, right? No accident that, but - we'd need a common interface to all those classes, and there isn't one - yet. Right now the `it` in all those 'factories' varies over incompatible types. Studio even notices some of those as? will never succeed and the compiler probably optimizes away the whole line...

And there you have it. My mental model is wishing for that loop, and the code would exist only once, and then moving parts into the helper class (at the moment pure data with just one additional partial-copy kind of constructor) would tend to reduce, not improve readability. So - the other way didn't even occur to me. Hmmm... I haven't yet solved all the impediments on the way to that loop, so... And then my personal dislike of code relying on Any... Can we get away with a constructor signature `CivilopediaEntry (name: String, description: String, category: CivilopediaCategories, object: INamed)` for now?

Suggestion: Let's agree that part isn't our final word, and revisit later. Discuss interface hierarchy elsewhere? At the moment the new abstraction tool is named `ICivilopediaText` because its goal is to enable text handling - but it could be a subclass to INamed making it.name easier to access, and it could abstract the common getImage instead of that CivilopediaCategories delegates kludge, and it could abstract the ruleset object source enumerators. Then it would be a `ICivilopediaObject`, and remain as only necessary parameter to the CivilopediaEntry constructor. But where in the 'inheritance tree' would it be in relation to NamedStats (which isn't an interface)?